target/linux/ramips/base-files/lib/preinit/07_set_preinit_iface_ramips

```diff
@@ -38,20 +38,20 @@ ramips_switchdev_from_board() {
 	local board=$(board_name)
 
 	case "$board" in
+	"tplink,archer-c5-v4")
+		switchdev=rtl8367s
+		;;
 	esac
 
 	echo "$switchdev"
 }
 
 ramips_set_preinit_iface() {
-	local switchdev=$(ramips_switchdev_from_board)
-	[ -n "$switchdev" ] || switchdev=$(ramips_switchdev_from_soc)
-
-	if [ -n "$switchdev" ]; then
+	for switchdev in $(ramips_switchdev_from_board) $(ramips_switchdev_from_soc); do
```

Comment:
The result we need is to reset both switches (as the code does), I'm not sure if it is better to simply reset both SoC and board switches or let board-specific switches be a list of switches, something like:

```sh
	case "$board" in
	"tplink,archer-c5-v4")
		switchdev="mt7620 rtl8367s"
		;;
	esac
```
This way, if needed, a board could avoid SoC switch to be reseted.

I'll let someone else comment on this.